pkg/install/app/vaultserverbuiltinconfigdeps.go

```diff
@@ -0,0 +1,351 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	appsv1obj "github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/api/appsv1"
+	corev1obj "github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/api/corev1"
+	rbacv1obj "github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/api/rbacv1"
+	"github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/helper"
+	"github.com/puppetlabs/leg/k8sutil/pkg/controller/obj/lifecycle"
+	"github.com/puppetlabs/leg/k8sutil/pkg/norm"
+	"github.com/puppetlabs/relay-core/pkg/apis/install.relay.sh/v1alpha1"
+	"github.com/puppetlabs/relay-core/pkg/model"
+	"github.com/puppetlabs/relay-core/pkg/obj"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	VaultConfigFileName   = "vault.hcl"
+	VaultConfigVolumeName = "vault-config"
+	VaultConfigVolumePath = "/var/run/vault/config"
+	VaultDataVolumePath   = "/vault/data"
+	VaultDataVolumeName   = "data"
+	VaultIdentifier       = "vault"
+)
+
+type VaultServerBuiltInConfigDeps struct {
+	Core               *obj.Core
+	ClusterRoleBinding *rbacv1obj.ClusterRoleBinding
+	OwnerConfigMap     *corev1obj.ConfigMap
+	Service            *corev1obj.Service
+	ServiceAccount     *corev1obj.ServiceAccount
+	StatefulSet        *appsv1obj.StatefulSet
+	Labels             map[string]string
+}
+
+func (vd *VaultServerBuiltInConfigDeps) Load(ctx context.Context, cl client.Client) (bool, error) {
+	key := client.ObjectKey{
+		Name:      VaultIdentifier,
+		Namespace: vd.Core.Key.Namespace,
+	}
+
+	vd.OwnerConfigMap = corev1obj.NewConfigMap(helper.SuffixObjectKey(key, "owner"))
+
+	vd.ClusterRoleBinding = rbacv1obj.NewClusterRoleBinding(helper.SuffixObjectKey(key, "token-reviewer").Name)
+	vd.Service = corev1obj.NewService(key)
+	vd.ServiceAccount = corev1obj.NewServiceAccount(key)
+	vd.StatefulSet = appsv1obj.NewStatefulSet(key)
+
+	ok, err := lifecycle.Loaders{
+		vd.OwnerConfigMap,
+		vd.ClusterRoleBinding,
+		vd.Service,
+		vd.ServiceAccount,
+	}.Load(ctx, cl)
+	if err != nil {
+		return false, err
+	}
+
+	return ok, nil
+}
+
+func (vd *VaultServerBuiltInConfigDeps) Owned(ctx context.Context, owner lifecycle.TypedObject) error {
+	return helper.Own(vd.OwnerConfigMap.Object, owner)
+}
+
+func (vd *VaultServerBuiltInConfigDeps) Persist(ctx context.Context, cl client.Client) error {
+	if err := vd.OwnerConfigMap.Persist(ctx, cl); err != nil {
+		return err
+	}
+
+	os := []lifecycle.Ownable{
+		vd.Service,
+		vd.ServiceAccount,
+		vd.StatefulSet,
+	}
+
+	for _, o := range os {
+		if err := vd.OwnerConfigMap.Own(ctx, o); err != nil {
+			return err
+		}
+	}
+
+	objs := []lifecycle.Persister{
+		vd.ClusterRoleBinding,
```

Comment:
Since this can't be owned by the config map, maybe you want to just make it owned by the RelayCore itself (assuming they're both not namespaced, I think)?

Reply:
`RelayCore` is namespaced at the moment. We need a general pattern/solve for this (and it needs applied to all of the other components, as we have several instances of this already). And then obviously a solution for handling proper deletion of all of these resources. That can be handled as part of a separate issue.